include/swift/RemoteInspection/ReflectionContext.h

```diff
@@ -1040,8 +1040,10 @@ class ReflectionContext
       if (auto Meta = readMetadata(*MetadataAddress)) {
         auto GenericHeapMeta =
           cast<TargetGenericBoxHeapMetadata<Runtime>>(Meta.getLocalBuffer());
-        return getMetadataTypeInfo(GenericHeapMeta->BoxedType,
-                                   ExternalTypeInfo);
+
+        auto MetadataAddress = GenericHeapMeta->BoxedType;
+        auto TR = readTypeFromMetadata(MetadataAddress);
```

Comment:
getMetadataTypeInfo() also checks that TR was not nullptr first, do we need to do that here?